### PR TITLE
Added support for IP filtering rule commands in IotHub

### PR DESCRIPTION
**Pull request by rajeevmv**
The content to be added to Changelog is as follows:

Added support for the following IP filter-rules commands in IotHub:
1. List IP filter rules: azure iothub ipfilter-rules list [resource-group] [name]
2. Set IP filter rules: azure iothub ipfilter-rules set [resource-group] [name] [input-file]

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@rajeevmv__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ravokkar). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/arm/iothub/iothub._js [hunk 1]
```diff
@@ -484,6 +484,148 @@ exports.init = function (cli) {
...
+        iothubDescription = client.iotHubResource.get(resourceGroup, name, _);
+      } finally {
+        progress.end();
+      }
+
+      ////////////////////////////////////////
+      // Output the result.                 //
+      ////////////////////////////////////////
+	 
+	  if (iothubDescription) {
```
Comment (amarzavery) [inline lib/commands/arm/iothub/iothub._js]: indentation should be 2 spaces.

File: lib/commands/arm/iothub/iothub._js [hunk 2]
```diff
@@ -484,6 +484,148 @@ exports.init = function (cli) {
...
+      }
+
+      ////////////////////////////////////////
+      // Output the result.                 //
+      ////////////////////////////////////////
+	 
+	  if (iothubDescription) { 
+		if (options.outputFile) {
+			var fs = require('fs');
+			fs.writeFileSync(options.outputFile, JSON.stringify(iothubDescription.properties.ipFilterRules, null, 2));
```
Comment (amarzavery) [inline lib/commands/arm/iothub/iothub._js]: surround it with a try catch and throw a better error message.
2 reasons
1. Whenever you use sync version of a method it should be surrounded with try catch.
2. Whenever you use JSON.stringify and JSON.patch it should be surrounded with try catch

File: lib/commands/arm/iothub/iothub._js [hunk 3]
```diff
@@ -484,6 +484,148 @@ exports.init = function (cli) {
...
+				row.cell($('IpMask'), rule.ipMask);
+			});
+        });
+      }
+    });
+
+  /**
+   * Set the Iot Hub IP Filter Rules
+   */
+  iotHubIpFilterRules.command('set [resource-group] [name]')
```
Comment (amarzavery) [inline lib/commands/arm/iothub/iothub._js]: [input-file] missing as a positional parameter

File: lib/commands/arm/iothub/iothub._js [hunk 4]
```diff
@@ -484,6 +484,148 @@ exports.init = function (cli) {
...
+  /**
+   * List the Iot Hub IP Filter Rules
+   */
+  iotHubIpFilterRules.command('list [resource-group] [name]')
+    .description($('List the IoT hub IP filter rules'))
+    .usage('<resourceGroup> <name> [options]')
+    .option('-g --resource-group <resourceGroup>', $('The name of the resource group'))
+    .option('-n --name <name>', $('The name of the IoT hub'))
+    .option('-s --subscription [subscription]', $('The subscription identifier'))
+	.fileRelatedOption('-f --output-file [output-file]', $('the path to the JSON-formatted output IP filter rules file in the file system'))
```
Comment (amarzavery) [inline lib/commands/arm/iothub/iothub._js]: indentation. replace tabs with 2 spaces

Comment (amarzavery) [inline lib/commands/arm/iothub/iothub._js]: Can you show an example of the JSON content that should be present in the file? How will the user know what to put it  in the file?

File: lib/commands/arm/iothub/iothub._js [hunk 5]
```diff
@@ -484,6 +484,148 @@ exports.init = function (cli) {
...
+        progress.end();
+      }
+
+	  ////////////////////////////////////////
+      // Setting the properties.            //
+      ////////////////////////////////////////
+
+	  if (iothubDescription) {  	  
+		var fs = require('fs');
+		var jsonFile = fs.readFileSync(options.inputFile);
```
Comment (amarzavery) [inline lib/commands/arm/iothub/iothub._js]: try catch

**Comment by amarzavery** [comment]
@rajeevmv - Please make sure to add proper content in the description of the PR. Whatever you add there will go into the Changelog of the release. Make sure to add sentences in past tense and it should start with a verb (Added support for IP filtering in iothub foo bar command)

**Pull request merged**